features/build_runner/build_runner.rb

```diff
@@ -1,11 +1,185 @@
 module FastlaneCI
-  # Abstract class that represents a build runner, used
+  # Class that represents a build runner, used
   # to run tests for a given commit sha
+  #
+  # Responsible for
+  # - Loading up _fastlane_ and running a lane with it, checking the return status
+  # - Raise an exception if build fails, with information that can be handled by `TestRunnerService`
+  # - Reporting back a list of artifacts  to `TestRunnerService`
+
   class BuildRunner
+    include FastlaneCI::Logging
+
+    # Reference to the FastlaneCI::Project of this particular build run
+    attr_accessor :project
+
+    # The code hosting service we want to report the status back to
+    # TODO: this probably shouldn't be stored here, think about using
+    # 	callbacks or similar
+    attr_accessor :code_hosting_service
+
+    # A reference to FastlaneCI::Build
+    attr_accessor :current_build
+
+    # The commit sha we want to run the build for
+    attr_accessor :sha
+
+    # All lines that were generated so far, this might not be a complete run
+    # This is an array of hashes
+    # TODO: have a class representing a Row (has to offer dynamic values though, as we might have non fastlane runners in the future)
+    attr_accessor :all_build_output_log_lines
+
+    # All blocks listening to changes for this build
+    attr_accessor :build_change_observer_blocks
+
+    def initialize(project: nil, sha: nil, github_service: nil)
+      self.project = project
+      self.sha = sha
+
+      self.all_build_output_log_lines = []
+      self.build_change_observer_blocks = []
+
+      # TODO: provider credential should determine what exact CodeHostingService gets instantiated
+      self.code_hosting_service = github_service
+
+      self.prepare_build_object
+    end
+
+    # Use this method for additional setup for subclasses
+    def setup
+      not_implemented(__method__)
+    end
+
+    # Starts the build, incrementing the build number from the number of builds
+    # for a given project
+    #
+    # @return [nil]
+    def start
+      start_time = Time.now
+
+      logger.debug("Running runner now")
```

Comment:
Let's add some identifiable information here so we know what runner this is

Reply:
Yesss